### DISCUSSION TITLE
Download ZIP points to wrong url when not on master branch

### DISCUSSION THREAD

**Discussion by ayanamist** [community/community#23852] [New to GitHub]
<aside class="onebox githubfolder" data-onebox-src="https://github.com/Dreamacro/clash-dashboard/tree/gh-pages">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/Dreamacro/clash-dashboard/tree/gh-pages" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h3><a href="https://github.com/Dreamacro/clash-dashboard/tree/gh-pages" target="_blank" rel="noopener nofollow ugc">GitHub - Dreamacro/clash-dashboard at gh-pages</a></h3>

  <p><a href="https://github.com/Dreamacro/clash-dashboard/tree/gh-pages" target="_blank" rel="noopener nofollow ugc">gh-pages</a></p>

  <p><span class="label1">web port of clash. Contribute to Dreamacro/clash-dashboard development by creating an account on GitHub.</span></p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/d/bd0b6c555519a1e0548cec84d006f91058ab5bcb.png" data-download-href="/uploads/short-url/qYmIo92OpoanG5CtfLOvSIlZB5p.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/406779/181079708-f258ecb4-b93a-411c-91c8-72db0037e96a.png" alt="image" data-base62-sha1="qYmIo92OpoanG5CtfLOvSIlZB5p" width="645" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd0b6c555519a1e0548cec84d006f91058ab5bcb_2_645x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd0b6c555519a1e0548cec84d006f91058ab5bcb_2_967x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd0b6c555519a1e0548cec84d006f91058ab5bcb_2_1290x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd0b6c555519a1e0548cec84d006f91058ab5bcb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1738×1346 200 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>It’s on <code>gh-pages</code> branch, but “Download ZIP” url still points to the zip file url containing files of master branch, but not of current branch like <code>https://github.com/Dreamacro/clash-dashboard/archive/refs/heads/gh-pages.zip</code></p>

**Comment by nazimboudeffa** [comment]
<p>Try with this link</p>
<p><a href="https://github.com/Dreamacro/clash-dashboard/zipball/gh-pages" rel="noopener nofollow ugc">https://github.com/Dreamacro/clash-dashboard/zipball/gh-pages</a></p>
<p>by adding <strong>zipball</strong> in the url</p>

**Comment by ayanamist** [comment]
<p>I know the workaround, i have already written down the correct url, i just want to report bug.</p>

**Comment by JellySnek** [comment]
<p>I can confirm this here. Same issue. No matter which branch I am on, it always downloads main.</p>

**Comment by bobanum** [comment]
<p>Same here. As a teacher, it’s really ennoying to me. Is there an official bug reporting process for Github?</p>

**Comment by GuteLaune** [comment, marked as answer]
<p>It works again<br>
:tada:</p>
<p><span class="spoiler"></span></p>
